tests/testing.ts

```diff
@@ -275,6 +283,74 @@ export function runTests(force: boolean, onlyResults: boolean, unsafe = false, e
   }
 }
 
+function isRustInstalled() {
+  try {
+    execSync('rustc --version', { stdio: 'inherit' });
+    execSync('cargo --version', { stdio: 'inherit' });
+    execSync('rustup --version', { stdio: 'inherit' });
+  } catch (e) {
+    return false;
+  }
+  return true;
+}
+
+function cairo1setup(): boolean {
+  // check if rust is installed
+  if (!isRustInstalled()) {
+    console.log(
+      "Please make sure you've installed rust alongside with rustup and cargo from here: https://www.rust-lang.org/tools/install",
+    );
+    return false;
+  }
+
+  // check if cairo1 directory exists and setup has been done
+  if (
+    !fs.existsSync(CAIRO1_DIR) ||
+    !fs.existsSync(CAIRO1_CARGO_TOML) ||
+    !fs.existsSync(CAIRO1_COMPILE_BIN) ||
+    !fs.existsSync(SIERRA_RUN_BIN)
+  ) {
+    console.log(
+      "Please make sure you've done the cairo1 setup from here: https://github.com/starkware-libs/cairo#readme",
+    );
+    return false;
+  }
+  return true;
+}
+
+function doCairo1Setup() {
+  if (!isRustInstalled) {
+    try {
+      console.log('Installing rust...');
+      execSync("curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh", {
+        stdio: 'inherit',
+      });
+      execSync('rustup override set stable && rustup update', { stdio: 'inherit' });
+    } catch (e) {
+      throw new Error(
+        'Failed to install rust, try manual installation of rust from https://www.rust-lang.org/tools/install',
+      );
+    }
+  }
```

Comment:
I think it is not the best approach to install rust for the users automatically. There multiple ways of installing a package, and this one could be unwanted